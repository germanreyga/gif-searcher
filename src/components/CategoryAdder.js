import React, { useState } from "react";
import PropTypes from "prop-types";

export const CategoryAdder = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState("");

  const handleSubmit = (event) => {
    event.preventDefault();

    if (inputValue.trim().length > 2) {
      setCategories((previousCategories) => [
        inputValue.trim(),
        ...previousCategories,
      ]);

      setInputValue("");
    } else {
      window.alert("You must add a valid name");
    }
  };

  return (
    <form onSubmit={handleSubmit}>
      <input
        type="text"
        placeholder="e.j. Hunter X Hunter"
        value={inputValue}
        onChange={(e) => setInputValue(e.target.value)}
      />

      <button type="submit">Add new category</button>
    </form>
  );
};

CategoryAdder.propTypes = {
  setCategories: PropTypes.func.isRequired,
};
