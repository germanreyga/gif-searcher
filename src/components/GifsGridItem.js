import React from "react";
import PropTypes from "prop-types";

export const GifsGridItem = ({ id, title, url }) => {
  return (
    <div className="gif-card animate__animated animate__fadeInLeft">
      <img src={url} alt={title} />
      <p>{title ? title : `No title for this GIF`}</p>
    </div>
  );
};

GifsGridItem.propTypes = {
  id: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};
