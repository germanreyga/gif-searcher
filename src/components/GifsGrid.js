import React from "react";
import { GifsGridItem } from "./GifsGridItem";
import { useFetchGifs } from "../hooks/useFetchGifs";
import PropTypes from "prop-types";

export const GifsGrid = ({ category }) => {
  const { data: imgs, loading } = useFetchGifs(category);

  return (
    <>
      <h3 className="category-name animate__animated animate__fadeIn">
        {category}
      </h3>
      {loading && (
        <p className="animate__animated animate__flash">"Loading images..."</p>
      )}
      {
        <div className="card-grid">
          {imgs.map((img) => {
            return <GifsGridItem key={img.id} {...img} />;
          })}
        </div>
      }
    </>
  );
};

GifsGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
