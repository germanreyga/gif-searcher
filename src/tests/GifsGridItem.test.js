import React from "react";
import { shallow } from "enzyme";
import "@testing-library/jest-dom";
import { GifsGridItem } from "../components/GifsGridItem";

describe("testing the <GifsGridItem/> component for the Gifs App", () => {
  test("should show the component correctly", () => {
    const wrapper = shallow(
      <GifsGridItem id={"0"} title="Random" url="www.google.com" />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
