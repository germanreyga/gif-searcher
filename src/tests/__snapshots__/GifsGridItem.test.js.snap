// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing the <GifsGridItem/> component for the Gifs App should show the component correctly 1`] = `
<div
  className="gif-card animate__animated animate__fadeInLeft"
>
  <img
    alt="Random"
    src="www.google.com"
  />
  <p>
    Random
  </p>
</div>
`;
